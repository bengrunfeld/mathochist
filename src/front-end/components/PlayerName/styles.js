import styled from 'styled-components'

const PlayerNameContainer = styled.div`
  padding-left: 45px !important;
`

const Title = styled.h5`
  font-size: 18px;
  font-weight: 300;
`

const Name = styled.p`
  font-size: 40px;
  color: #3c78d8;
  text-transform: uppercase;
  margin-left: -2px;
`

export { PlayerNameContainer, Title, Name }
