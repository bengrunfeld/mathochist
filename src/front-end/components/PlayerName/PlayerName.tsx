import React from 'react'
import PropTypes from 'prop-types'

import { Title, Name } from './styles'

const PlayerName = ({ playerName }) => (
  <div className='col-6'>
    <Title>Player:</Title>
    <Name>{playerName}</Name>
  </div>
)

PlayerName.defaultProps = {
  playerName: 'Beynish'
}

PlayerName.propTypes = {
  playerName: PropTypes.string
}

export default PlayerName
