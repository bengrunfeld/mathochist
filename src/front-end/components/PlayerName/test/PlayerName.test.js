import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import PlayerName from '../PlayerName'

describe('<PlayerName />', () => {
  test('matches the snapshot', () => {
    const tree = mount(<PlayerName />)
    expect(tree).toMatchSnapshot()
  })
})
