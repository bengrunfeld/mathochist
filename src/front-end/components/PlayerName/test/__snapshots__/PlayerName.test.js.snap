// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerName /> matches the snapshot 1`] = `
<PlayerName
  playerName="Beni"
>
  <styled.div
    className="col-6"
  >
    <StyledComponent
      className="col-6"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "fARYbS",
            "rules": Array [
              "
  padding-left: 45px !important;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="col-6 sc-bdVaJa fARYbS"
      >
        <styled.h5>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "jszCaR",
                  "rules": Array [
                    "
  font-size: 18px;
  font-weight: 300;
",
                  ],
                },
                "displayName": "styled.h5",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "h5",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h5
              className="sc-bwzfXH jszCaR"
            >
              Player:
            </h5>
          </StyledComponent>
        </styled.h5>
        <styled.p>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "ezsysF",
                  "rules": Array [
                    "
  font-size: 40px;
  color: #3c78d8;
  text-transform: uppercase;
  margin-left: -2px;
",
                  ],
                },
                "displayName": "styled.p",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <p
              className="sc-htpNat ezsysF"
            >
              Beni
            </p>
          </StyledComponent>
        </styled.p>
      </div>
    </StyledComponent>
  </styled.div>
</PlayerName>
`;
